config-object-store/src/main/java/org/hypertrace/config/objectstore/IdentifiedFilterPushedDownObjectStore.java

```diff
@@ -0,0 +1,42 @@
+package org.hypertrace.config.objectstore;
+
+import org.hypertrace.config.service.change.event.api.ConfigChangeEventGenerator;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc;
+import org.hypertrace.config.service.v1.Filter;
+import org.hypertrace.config.service.v1.Pagination;
+import org.hypertrace.config.service.v1.SortBy;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+
+import java.util.Collections;
+import java.util.List;
+
+public abstract class IdentifiedFilterPushedDownObjectStore<T, F> extends IdentifiedObjectStore<T> {
+
+  protected IdentifiedFilterPushedDownObjectStore(
+      ConfigServiceGrpc.ConfigServiceBlockingStub configServiceBlockingStub,
+      String resourceNamespace,
+      String resourceName,
+      ConfigChangeEventGenerator configChangeEventGenerator,
+      ClientConfig clientConfig) {
+    super(
+        configServiceBlockingStub,
+        resourceNamespace,
+        resourceName,
+        configChangeEventGenerator,
+        clientConfig);
+  }
+
+  public List<ContextualConfigObject<T>> getAllObjects(
+      RequestContext context, F filterInput, Pagination pagination) {
+    Filter filter = buildFilter(filterInput);
+    List<SortBy> sortByList = buildSortByList(filterInput);
+    return getAllObjectsWithFilter(context, filter, pagination, sortByList);
+  }
+
+  protected abstract Filter buildFilter(F filterInput);
+
+  /** Constructs a list of SortBy from client filter or returns an empty list. */
+  protected List<SortBy> buildSortByList(F filterInput) {
```

Comment:
- Sorts should be passed into the getAllObjects call. I would probably abstract it and let the callers specify, similar to filter. So a new type param `S` and accept a `List<S>` with a method like `protected abstract SortBy buildSort(S sortInput);`
- We'd also want to add equivalent implementations for all the other methods in `IdentifiedObjectStoreWithFilter` there's a plural and singular version of both get object and get data.
- Since it's a new class we can also fix the naming, it's not "getAll" since we're applying a filter, it should be "getMatching"

Reply:
Fixed all of them, please check now.